src/GafferImage/Display.cpp

```diff
@@ -268,43 +301,56 @@ class GafferDisplayDriver : public IECoreImage::DisplayDriver
 			Display::driverCreatedSignal()( driver.get(), parameters.get() );
 		}
 
-		ConstFloatVectorDataPtr getTile( const V2i &tileOrigin, size_t channelIndex )
+		struct Tile
 		{
-			V2i tileIndex = tileOrigin / ImagePlug::tileSize();
-
-			if(
-				tileIndex.x < m_tiles.index_bases()[0] ||
-				tileIndex.x >= (int)(m_tiles.index_bases()[0] + m_tiles.shape()[0] ) ||
-				tileIndex.y < m_tiles.index_bases()[1] ||
-				tileIndex.y >= (int)(m_tiles.index_bases()[1] + m_tiles.shape()[1] )
-			)
+			Tile(): backBufferData( ImagePlug::blackTile()->copy() ), backBuffer( backBufferData->writable() ), dirty( false ), cachedTile( ImagePlug::blackTile() ), cachedForDataCount( 0 ), updateCount( 0 )
 			{
-				// outside data window
-				return nullptr;
 			}
 
-			tbb::spin_rw_mutex::scoped_lock tileLock( m_tileMutex, false /* read */ );
+			Tile( const Tile& other ) : Tile()
+			{
+				*this = other;
+			}
 
-			ConstFloatVectorDataPtr result = m_tiles[tileIndex.x][tileIndex.y][channelIndex];
-			if( !result )
+			Tile& operator=( const Tile& other )
 			{
-				result = ImagePlug::blackTile();
+				memcpy( &backBuffer[0], &other.backBuffer[0], backBuffer.size() * sizeof( float ) );
+				dirty = (bool)other.dirty;
+				cachedTile = other.cachedTile;
+				cachedForDataCount = other.cachedForDataCount;
+				return *this;
 			}
 
-			return result;
-		}
+			FloatVectorDataPtr backBufferData;
+			std::vector<float> &backBuffer;
+			std::atomic<bool> dirty;
+			tbb::spin_rw_mutex mutex;
 
-		void setTile( const V2i &tileOrigin, size_t channelIndex, ConstFloatVectorDataPtr tile )
+			// Use mutex to access these 2
+			ConstFloatVectorDataPtr cachedTile;
+			int cachedForDataCount;
+			int updateCount;
```

Comment:
I can't see how we are using `updateCount`. We seem to increment it but not actually do anything else. Am I missing something?

Reply:
Removed.